src/app/core/workflow/flow.js

```diff
@@ -16,21 +16,26 @@ function Flow() {
   };
   //start workflow
   this.start = function(workflow) {
-    d = $.Deferred();
-    if (counter > 0) {
-      console.log("reset workflow before restarting");
-    }
-    _workflow = workflow;
-    inputs = workflow.getInputs();
-    context = workflow.getContext();
-    steps = workflow.getSteps();
-    // check if there are steps
-    if (steps && steps.length) {
-      //run step (first)
-      this.runStep(steps[0], inputs, context);
-    }
-    // return a promise that will be reolved if all step go right
-    return d.promise();
+    return new Promise((resolve, reject) => {
+      //** Assign d module variable to an object having resolve, reject methods used on other module method
```

Comment:
Specific code comments (jsdoc, but not only those) on the usefulness of a variable must be written when declaring it otherwise your development IDE is not able to recognize them, that is:

```
/**
 * Module variable having resolve, reject methods used on other module method
 */
let d;